### ISSUE TITLE
train/valid probs

### ISSUE THREAD

**Issue by danpovey** [waldo-seg/waldo#63]
Guys, 
We should have a mechanism to compute either accuracies on a subset of training data, or objective function values on validation data.  (Are we doing this already?)
This will show whether our model is underfitting or overfitting, which right now I have no idea about.

**Comment by YiwenShaoStephen** [comment]
train.py will give you the BCE loss on train and val data. If you want to further see the segmentation results on train and val, you need to run segment.py on them on using the scoring.py to get the MAP (mean average precision)

**Comment by hhadian** [comment]
But, the segmentation algorithm also produces a final logprob for each
image. I guess it would be helpful to write that to disk too (and maybe
also its average on a whole test set).

**Comment by danpovey** [comment]
OK.  How different are the BCE losses for train and val, in the DBS2018
setup?

**Comment by YiwenShaoStephen** [comment]
Almost the same. So there is no overfitting yet. I think heavier image augmentation will help since the dataset is relatively small.

**Comment by danpovey** [comment]
Hossein, that's a good point.  Perhaps we could produce a log line per
file, summarizing various stats relating to the segmentation.   I assume
the stderr (and probably stdout too) of the segmenter code gets put in a
log file.

Yiwen, regarding the train/valid objective values: I think you can safely
increase the number of parameters in the model until you see overfitting,
and at that point start to worry about image augmentation.  (I see image
augmentation as primarily a way to reduce overfitting by artificially
expanding the amount of training data).

**Comment by YiwenShaoStephen** [comment]
OK, will try it.

**Comment by hhadian** [comment]
Will do it.
